test/distributed/org/apache/cassandra/distributed/test/PutGetTest.java

```diff
@@ -0,0 +1,306 @@
+        Assert.assertEquals(numPartitions * numRows, rows.length);
+        for (int k = 0; k < numPartitions; k++)
+        {
+            for (int c = 0; c < numRows; c++)
+            {
+                Object[] row = rows[k * numRows + c];
+                validateRow(row, numRows, k, c);
+            }
+        }
+    }
+
+    /**
+     * Simple put/get on a single partition with multiple rows, reading with slice queries.
+     * <p>
+     * Migrated from Python dtests putget_test.py:TestPutGet.test_wide_row().
+     */
+    @Test
+    public void testSliceQuery()
+    {
+        int numRows = 100;
+
+        writeRows(1, numRows);
+
+        String query = "SELECT * FROM %s WHERE k=? AND c>=? AND c<?";
+        for (int sliceSize : Arrays.asList(10, 20, 100))
+        {
+            for (int c = 0; c < numRows; c = c + sliceSize)
+            {
+                Object[][] rows = readRows(query, 0, c, c + sliceSize);
+                Assert.assertEquals(sliceSize, rows.length);
+
+                for (int i = 0; i < sliceSize; i++)
+                {
+                    Object[] row = rows[i];
+                    validateRow(row, numRows, 0, c + i);
+                }
+            }
+        }
+    }
+
+    /**
+     * Simple put/get on multiple partitions with multiple rows, reading with IN queries.
+     */
+    @Test
+    public void testInQuery()
+    {
+        int numPartitions = 10;
+        int numRows = 10;
+
+        writeRows(numPartitions, numRows);
+
+        String query = "SELECT * FROM %s WHERE k IN (?, ?)";
+        for (int k = 0; k < numPartitions; k += 2)
+        {
+            Object[][] rows = readRows(query, k, k + 1);
+            Assert.assertEquals(numRows * 2, rows.length);
+
+            for (int i = 0; i < 2; i++)
+            {
+                for (int c = 0; c < numRows; c++)
+                {
+                    Object[] row = rows[i * numRows + c];
+                    validateRow(row, numRows, k + i, c);
+                }
+            }
+        }
+    }
+
+    /**
+     * Writes {@code numPartitions} with {@code numRows} each, with overrides in different sstables and memtables.
+     */
+    private void writeRows(int numPartitions, int numRows)
+    {
+        String update = format("UPDATE %s SET v=? WHERE k=? AND c=?");
+        ICoordinator coordinator = cluster.coordinator(1);
+        for (int k = 0; k < numPartitions; k++)
+        {
+            // insert all the partition rows in a single sstable
+            for (int c = 0; c < numRows; c++)
+                coordinator.execute(update, QUORUM, c, k, c);
+            cluster.forEach(i -> i.flush(KEYSPACE));
+
+            // override some rows in a second sstable
+            for (int c = 0; c < numRows; c += 2)
+                coordinator.execute(update, QUORUM, c + numRows, k, c);
+            cluster.forEach(i -> i.flush(KEYSPACE));
+
+            // override some rows only in memtable
+            for (int c = 0; c < numRows; c += 5)
```

Comment:
nit: It might be nice to make constants out of the skip intervals 2 and 5 (so at least we can reuse it below during validation).